packages/caip/src/typeGuards.ts

```diff
@@ -21,20 +16,21 @@ export const isChainReference = (
 export const isAssetNamespace = (
   maybeAssetNamespace: AssetNamespace | string,
 ): maybeAssetNamespace is AssetNamespace =>
-  ASSET_NAMESPACE_STRINGS.some((s) => s === maybeAssetNamespace)
+  Object.values(ASSET_NAMESPACE).some((s) => s === maybeAssetNamespace)
 
 export const isAssetReference = (
   maybeAssetReference: AssetReference | string,
 ): maybeAssetReference is AssetReference =>
   Object.values(ASSET_REFERENCE).some((s) => s === maybeAssetReference)
 
 export const isAssetId = (maybeAssetId: AssetId | string): maybeAssetId is AssetReference => {
-  const matches = parseAssetIdRegExp.exec(maybeAssetId)?.groups
+  const matches = parseAssetIdRegExp.exec(maybeAssetId)
+  const [, chainNamespace, chainReference, assetNamespace] = matches || []
```

Comment:
We also can't do `|| []` because the type guards and associated type guard assertions do not handle `undefined` values, which the destructured values would be if it was indeed an empty array.

TypeScript misses this when using an array constant on the right-hand side of a logical or operator, though you can confirm the downstream effects if you, say, try passing `undefined` to `isChainNamespace` on line 31.

One option is to replace line 28 with the following code that includes an early exit, as `matches ` will either be a valid array, or `null`:

```ts
if (!matches) return false
const [, chainNamespace, chainReference, assetNamespace] = matches
```

Then remove the truthy check for `matches` on line 30, as by this stage we know it's truthy.

We can also tighten up the array destructuring by using object destructuring syntax on the array's numeric keys:

```ts
const { 1: chainNamespace, 2: chainReference, 3: assetNamespace } = matches
```